### ISSUE TITLE
The --use-element-hierarchy parameter for conversion .ifc in .glb

### ISSUE THREAD

**Issue by freeinform** [IfcOpenShell/IfcOpenShell#790]
Whether it will be possible to use the --use-element-hierarchy parameter for conversion .ifc in .glb?
Now it is only applicable for .dae. For GLTF, it would also be very useful.
Since we need to keep the hierarchy in GLTF, we currently have to convert according to the scheme: .ifc -> .dae using --use-element-hierarchy, and then .dae -> .glb using COLLADA2GLTF.

And it would be very elegant if it was possible, optionally, through a parameter, when converting .ifc to .glb, write to GLTF node.extras mesh.extras IFC parameters IfcPropertySet as an object and IFC type (for example: IfcWallStandardCase etc.). Or at least write node.extras.GlobalId or mesh.extras.GlobalId (IFC Global Id), then we can take any data from the decomposition. That is, leave the connection between IFC and GLTF.

This will be very useful for applications that use GLTF for IFC visualization.

**Comment by aothms** [comment]
> --use-element-hierarchy parameter for conversion .ifc in .glb?

I agree, would be useful, but not sure if I can find the time for this.

> write to GLTF node.extras mesh.extras IFC parameters IfcPropertySet as an object and IFC type 

What I usually recommend is to use the converted XML (also from IfcConvert) alongside the generated DAE/glTF/OBJ. Adding all the parameters will blow up the size of the file significantly. The identifiers in the glTF are currently an encoding (hex rfc 4122) of the IFC GlobalId already so you can cross reference the files. You can also use `--use-element-guids` to have the original base64 IFC encoding directly in the glTF.

**Comment by vegarringdal** [comment]
any chance we will be able to get this ?

**Comment by aothms** [comment]
Since there is a workaround (COLLADA2GLTF) it's not immensely high on my todo list, but if there's somebody else willing to work on this I'd happily assist

**Comment by BIMvoice** [comment]
Status check on this request, verified against current v0.8.0:

**Hierarchy for glTF: implemented.** `--element-hierarchy` (current name of `--use-element-hierarchy`) now supports `.glb` output and produces a real nested node tree matching the IFC spatial structure (verified on a multi-storey model: site → building → storeys → walls). While verifying, we found the intermediate group nodes (storey/building/site) were misnamed — they inherited a leaf element's identifier instead of their own — fixed in #8771.

**`node.extras`/`mesh.extras` metadata: not implemented.** What exists instead is what @aothms described in 2020: node names encode the IFC GlobalId (hex/uuid by default, the literal base64 GlobalId with `--use-element-guids`, verified byte-for-byte), so cross-referencing a sidecar XML/JSON export by GlobalId works today. Embedding property-set data directly into glTF extras remains open for contribution given the file-size concern raised above.

This comment was written with AI assistance.

**Comment by BIMvoice** [comment]
Update: the `node.extras` half is now also implemented in #8771 as an opt-in `--gltf-extras` flag — off by default so file size is unchanged unless requested, writing `{GlobalId, IfcType, Name}` on every node including the hierarchy group nodes. @aothms's size concern is preserved by the default; the self-describing glb this issue asked for is available by opting in.

This comment was written with AI assistance.
